[stable] Fix up 2.087 C++ header regressions

kinke opened the pull request:
At least those I've encountered so far, incl. a cherry-pick of #10065.

SSoulaimane committed:
Update C++ headers - Follow up to #9282

kinke committed:
Fix up 2.087 C++ header regressions

ibuclaw reviewed (dismissed):
Please add tests, and make it harder for people to blindly break this in future.

kinke committed:
Improve string representation of SymOffExp offset

It's a 64-bit value, so don't ignore the 32 upper bits, and print it as
signed value for easily readable small negative offsets.

kinke committed:
Extend C++ interop tests by some virtuals & field offsets

thewilsonator commented:
Mecca failure unrelated.
